icevision/core/record_components.py

```diff
@@ -116,7 +116,11 @@ def set_img(self, img: np.ndarray):
         self.composite.set_img_size(ImgSize(width=width, height=height), original=True)
 
     def _repr(self) -> List[str]:
-        return [f"Image: {self.img}"]
+        if self.img is not None:
+            height, width, dims = self.img.shape
+            return [f"Image: {width}x{height}x{dims} <np.ndarray> Image"]
```

Comment:
Hey, tricky question: does this work with grayscale images? (I'm not sure they will have the last dimension)

Reply:
Oh, good one. It won't work there. So, shall I do another check in here like so:

```python
if len(self.img.shape)==3:
    ...
elif len(self.img.shape)==2:
    ...
```

Can't think of another method, because there's no way of knowing the type of image otherwise. 


**RELATED:**
In the previous method (above the one highlighted) - `set_img`, there's also a line that assumes 3 dims:
```python
    def set_img(self, img: np.ndarray):
        self.img = img
        height, width, _ = self.img.shape
```

Reply:
> In the previous method (above the one highlighted) - set_img, there's also a line that assumes 3 dims:
Hahah that would break for grayscale then as well

I have a suggestion to use instead of the if statements, but I don't know if this is actually better than use the if statements:
```python
rgb_img = np.ones((16, 16, 3))
gray_img = np.ones((16,16))

img = gray_img
height, width, channels = img.reshape(*img.shape[:2], -1).shape
```

Reply:
Interesting.
I checked - if statements are faster (and to me, personally, clearer too)

<img width="1008" alt="Screenshot 2021-03-18 at 12 11 39 PM" src="https://user-images.githubusercontent.com/17111474/111583893-1a176980-87e3-11eb-8c38-fc26fb5c9ee1.png">

Reply:
What an amazing theme, it seems like code from https://carbon.now.sh/ hahahahha

Let's go with the if statements then. Just add something like:
```python
raise ValueError("Expected image to have 3 or 2 dimensions but got {len(img.shape)}")
```

Feel free to change the error message to something clearer.

Reply:
Thanks. You can get the theme here [here](https://github.com/rsomani95/jupyter-custom-theme/) btw

Got it. Update coming right up

Reply:
Pushed in https://github.com/airctic/icevision/pull/705/commits/2a7f262d63774ef8bf975f9c84d44d74b153edca